runtime/interface.go

```diff
@@ -41,93 +43,110 @@ type Block struct {
 type ResolvedLocation = sema.ResolvedLocation
 type Identifier = ast.Identifier
 type Location = common.Location
+type AddressLocation = common.AddressLocation
 
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
-	// GetCode returns the code at a given location
-	GetCode(location Location) ([]byte, error)
-	// GetCachedProgram attempts to get a parsed program from a cache.
-	GetCachedProgram(Location) (*ast.Program, error)
-	// CacheProgram adds a parsed program to a cache.
-	CacheProgram(Location, *ast.Program) error
```

Comment:
I removed these two

Reply:
We have to figure out how to do caching properly. We don't want to remove it, but make it work and extend it to other objects, e.g. elaboration results from the checker.

Reply:
I have some ideas, I'll create a separate handler for caching for now until we discuss it.